File: backend/internal/models/user_model.go
package models

import "time"

type UserModel struct {
	ID                     int       `json:"id"`
	Role                   string    `json:"role_id"`
	Name                   string    `json:"name"`
	Country                string    `json:"country_id"`
	University             string    `json:"university"`
	Email                  string    `json:"email"`
	LeetCode               string    `json:"leetcode"`
	Codeforces             string    `json:"codeforces"`
	GitHub                 string    `json:"github"`
	Photo                  string    `json:"photo"`
	PreferredLanguage      string    `json:"preferred_language"`
	HackerRank             string    `json:"hackerrank"`
	Group                  string    `json:"group_id"`
	Phone                  string    `json:"phone"`
	TelegramUsername       string    `json:"telegram_username"`
	TelegramUID            string    `json:"telegram_uid"`
	LinkedIn               string    `json:"linkedin"`
	StudentID              string    `json:"student_id"`
	ShortBio               string    `json:"short_bio"`
	Instagram              string    `json:"instagram"`
	Birthday               time.Time `json:"birthday"`
	CV                     string    `json:"cv"`
	JoinedDate             time.Time `json:"joined_date"`
	ExpectedGraduationDate time.Time `json:"expected_graduation_date"`
	MentorName             string    `json:"mentor_name"`
	TShirtColor            string    `json:"tshirt_color"`
	TShirtSize             string    `json:"tshirt_size"`
	Gender                 string    `json:"gender"`
	CodeOfConduct          string    `json:"code_of_conduct"`
	Password               string    `json:"password"`
	CreatedAt              time.Time `json:"created_at"`
	UpdatedAt              time.Time `json:"updated_at"`
	Config                 string    `json:"config"`
	Department             string    `json:"department"`
	Inactive               bool      `json:"inactive"`
	FirstLogin             bool      `json:"firstlogin"`
}

type UserProfileResponse struct {
	ID                     int       `json:"id"`
	Name                   string    `json:"name"`
	Email                  string    `json:"email"`
	Photo                  string    `json:"photo"`
	University             string    `json:"university"`
	Role                   string    `json:"role"`
	Country                string    `json:"country"`
	JoinedDate             time.Time `json:"joined_date"`
	ExpectedGraduationDate time.Time `json:"expected_graduation_date"`
	ShortBio               string    `json:"short_bio"`
	LeetCode               string    `json:"leetcode"`
	Codeforces             string    `json:"codeforces"`
	GitHub                 string    `json:"github"`
	Instagram              string    `json:"instagram"`
	Phone                  string    `json:"phone"`
	StudentID              string    `json:"student_id"`
	TelegramUsername       string    `json:"telegram_username"`
	Group                  string    `json:"group"`
	Department             string    `json:"department"`
}

File: a2svhub/lib/features/auth/data/model/user_model.dart
import '../../domain/entities/user_entity.dart';

class UserModel extends UserEntity {
  UserModel({
    required super.id,
    required super.name,
    required super.email,
    super.photo,
    super.university,
    super.role,
    super.country,
    super.joinedDate,
    super.expectedGraduationDate,
    super.shortBio,
    super.leetcode,
    super.codeforces,
    super.github,
    super.instagram,
    super.phone,
    super.studentId,
    super.telegramUsername,
    super.group,
    super.department,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json['id'],
      name: json['name'] ?? '',
      email: json['email'],
      photo: json['photo'],
      university: json['university'],
      role: json['role'],
      country: json['country'],
      joinedDate: json['joined_date'],
      expectedGraduationDate: json['expected_graduation_date'],
      shortBio: json['short_bio'],
      leetcode: json['leetcode'],
      codeforces: json['codeforces'],
      github: json['github'],
      instagram: json['instagram'],
      phone: json['phone'],
      studentId: json['student_id'],
      telegramUsername: json['telegram_username'],
      group: json['group'],
      department: json['department'],
    );
  }

  // ✅ Add this method
  UserEntity toEntity() {
    return UserEntity(
      id: id,
      name: name,
      email: email,
      photo: photo,
      university: university,
      role: role,
      country: country,
      joinedDate: joinedDate,
      expectedGraduationDate: expectedGraduationDate,
      shortBio: shortBio,
      leetcode: leetcode,
      codeforces: codeforces,
      github: github,
      instagram: instagram,
      phone: phone,
      studentId: studentId,
      telegramUsername: telegramUsername,
      group: group,
      department: department,
    );
  }
}
